src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -830,6 +841,42 @@ private void WriteChunk(Stream stream, PngChunkType type, byte[] data, int offse
             stream.Write(this.buffer, 0, 4); // write the crc
         }
 
+        private void ScaleDownFrom8BitArray(ReadOnlySpan<byte> source, Span<byte> result, int bits)
```

Comment:
There's something not right here. We're getting pixels from the end of a row overwriting pixels at the start.

![bpp1](https://user-images.githubusercontent.com/385879/46074384-56702e80-c17f-11e8-86b0-fc93a28c2916.png)

![bpp1](https://user-images.githubusercontent.com/385879/46074396-5c660f80-c17f-11e8-9ec2-476bc3405744.png)

Reply:
Fixed. Thanks @dlemstra for spotting my daft mistakes.